selenium_view.py

```diff
@@ -0,0 +1,634 @@
+        with LocatorDescriptor._seq_cnt_lock:
+            o._seq_id = LocatorDescriptor._seq_cnt
+            LocatorDescriptor._seq_cnt += 1
+        return o
+
+    def __init__(self, klass, *args, **kwargs):
+        self.klass = klass
+        self.args = args
+        self.kwargs = kwargs
+
+    def __get__(self, obj, type=None):
+        if obj is None:  # class access
+            return self
+
+        # Cache on LocatorDescriptor
+        if self not in obj._widget_cache:
+            obj._widget_cache[self] = self.klass(obj, *self.args, **self.kwargs)
+        return obj._widget_cache[self]
+
+    def __repr__(self):
+        if self.args:
+            args = ', ' + ', '.join(repr(arg) for arg in self.args)
+        else:
+            args = ''
+        if self.kwargs:
+            kwargs = ', ' + ', '.join(
+                '{}={}'.format(k, repr(v)) for k, v in self.kwargs.iteritems())
+        else:
+            kwargs = ''
+        return '{}({}{}{})'.format(type(self).__name__, self.klass.__name__, args, kwargs)
+
+
+class Browser(object):
+    """Equivalent of pytest_selenium - browser functions"""
+    def __init__(self, selenium, plugin_class=None):
+        self.selenium = selenium
+        plugin_class = plugin_class or DefaultPlugin
+        self.plugin = plugin_class(self)
+
+    @property
+    def browser(self):
+        return self
+
+    @property
+    def product_version(self):
+        raise NotImplementedError('You have to implement product_version')
+
+    @staticmethod
+    def _process_locator(locator):
+        if isinstance(locator, WebElement):
+            return locator
+        try:
+            return Locator(locator)
+        except TypeError:
+            if hasattr(locator, '__element__'):
+                return locator.__element__()
+            else:
+                raise
+
+    def elements(self, locator, parent=None, check_visibility=False):
+        self.plugin.ensure_page_safe()
+        locator = self._process_locator(locator)
+        # Get result
+        if isinstance(locator, WebElement):
+            result = [locator]
+        else:
+            # Get the direct parent object
+            if parent:
+                root_element = self.elements(parent)
+            else:
+                root_element = self.selenium
+            result = root_element.find_elements(*locator)
+
+        if check_visibility:
+            result = filter(self.is_displayed, result)
+
+        return result
+
+    def element(self, locator, *args, **kwargs):
+        try:
+            return self.elements(locator, *args, **kwargs)[0]
+        except IndexError:
+            raise NoSuchElementException('Could not find an element {}'.format(repr(locator)))
+
+    def perform_click(self):
+        ActionChains(self.selenium).click().perform()
+
+    def click(self, *args, **kwargs):
+        self.move_to_element(*args, **kwargs)
+        # and then click on current mouse position
+        self.perform_click()
```

Comment:
Actually I sticked the call to the "ensurer" in the element query function so :)